// Copyright 2015 The gRPC Authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.melon.account";
option java_outer_classname = "AccountProto";
option objc_class_prefix = "ACC";

package account;

service Account {
  rpc requestUserLogin (LoginRequest) returns (CodeReply) {}
  rpc requestUserSign (SignRequest) returns (CodeReply) {}
  rpc checkConnect (ConnectRequest) returns (CodeReply) {}
  rpc requestLogout (LogoutRequest) returns (CodeReply) {}
}

message LoginRequest {
  string phone = 1;
  string password = 2;
  string deviceId = 3;
}

message SignRequest {
  string phone = 1;
  string password = 2;
  string deviceId = 3;
}

message ConnectRequest {
  string deviceId = 1;
}

message LogoutRequest {
  string phone = 1;
}

message CodeReply {
  int32 code = 1;
  string msg = 2;
}



